pkg/agent/agent.go

```diff
@@ -50,13 +53,19 @@ func (a *Agent) Run(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	metricsService := metricsservice.NewMetricsService(metricsservice.Config{
+		Metrics: metrics,
+	})
 
 	cat, err := catalog.Load(ctx, catalog.Config{
 		Log: a.c.Log.WithField(telemetry.SubsystemName, telemetry.Catalog),
 		GlobalConfig: catalog.GlobalConfig{
 			TrustDomain: a.c.TrustDomain.Host,
 		},
 		PluginConfig: a.c.PluginConfigs,
+		HostServices: []common_catalog.HostServiceServer{
+			common_services.MetricsServiceHostServiceServer(metricsService),
```

Comment:
Note that `Agent` and `Server` have a slightly different pattern on bringing in `HostServices`. `Agent` wants it all declared directly for loading as a slice, whereas `Server` lets the catalog handle slices and just passes in the services themselves.
They should probably be made consistent in another PR.

Reply:
Yep, I agree we should settle on a consistent method here.

Reply:
Created #991 to track